src/SNLS_TrDLDenseG.h

```diff
@@ -197,239 +196,99 @@ class SNLSTrDlDenseG
          bool reject_prev = false ;
          //
          // data for sorting out the step
-         double nr2norm = 0.0 ;
-         double alpha = 0.0 ;
-         double norm_s_sd_opt = 0.0 ;
-         double norm_grad = 0.0 ;
-         double norm_grad_inv = 0.0 ;
-         double qa = 0.0, qb = 0.0 ;
-         double Jg2 = 0.0 ;
-         double res_cauchy = 0.0 ;
-         double nrStep[_nDim] ;
-         double grad[_nDim] ;
+         double nrStep[_nDim];
+         double grad[_nDim];
+         double delx[_nDim];
+         double Jg_2;
          //
          while ( _nIters < _maxIter ) {
             //
             _nIters += 1 ;
 
-            if ( !reject_prev ) {
-               //
-               // have a newly accepted solution point
-               // compute information for step determination
-
-               this->computeSysMult( Jacobian, residual, grad, true );
-               // used to keep :
-               //	ngrad[iX] = -grad[iX] ;
-               // 	nsd[iX]   = ngrad[iX] * norm_grad_inv ; 
-               
-               // find Cauchy point
-               //
+            // This is done outside this step so that these operations can be done with varying solve
+            // techniques such as LU/QR or etc...
+            if(!reject_prev) {
+               // So the LU solve does things in step which causes issues...
```

Comment:
I am not sure what "does things in step" means here. 

You are talking about the LU solve doing things in place, so that a standard matrix-vector multiple with the Jacobian no longer works?

Reply:
Yeah sorry should have worded this better. Yes, the LU solve is done in-place which causes issues when doing normal matvec type operations.